extended-utxo-spec/extended-utxo-specification.tex

```diff
@@ -0,0 +1,1212 @@
+        valid if the ledger $\lambda$ is empty
+        (that is, if it is the initial
+        transaction). Note that the details of this rule depend on the
+        monetary policy of the ledger itself, as there may be other
+        transactions that forge value. \red{Expand on this.  What other transactions?  Mining?}
+        }
+    \end{center}
+      
+    \item \label{rule:value-is-preserved} \textbf{Value is preserved}:
+    \[
+      t.\forge + \sum_{i \in t.\inputs} \getvalue(i, \lambda) = t.\fee + \sum_{o \in t.\outputs} o.\val.
+    \]
+    \item \label{rule:no-double-spending} \textbf{No output is double spent:}
+    \[
+     \textrm{If } i_1, i_2 \in t.\inputs \textrm{ and }  i_1.\mathit{outputRef} = i_2.\mathit{outputRef}
+     \textrm{ then } i_1 = i_2.
+    \]
+    \item\label{rule:all-inputs-validate} \textbf{All inputs validate:}
+    \[
+    \textrm{For all } i \in t.\inputs,\enspace \llbracket
+    i.\validator(\txout(i, \lambda).\datascript,i.\redeemer, \sigma(\lambda, t, i)) \rrbracket \ne \textsf{error}.
+      \]
+    \item\label{rule:validator-scripts-hash} \textbf{Validator scripts hash to their output addresses:}
+    \[
+      \textrm{For all } i \in t.\inputs,\enspace i.\validator^{\#} = \txout(i, \lambda).\addr.
+    \]
+\end{enumerate}
+\caption{$\sigma$-validity of a transaction $t$ in the XUTXO-1 model}
+\label{fig:xutxo-1-validity}
+\end{ruledfigure}
+
+
+
+\noindent In practice, $\sigma$-validity imposes a limit on the size
+of the $\mathsf{Script}$ values of the $\validator$, $\redeemer$ and
+$\datascript$ fields and the result of $\sigma$. The validation of a
+single transaction must take place within one slot, so the evaluation
+of $\llbracket ~ \rrbracket$ cannot take longer than one slot
+(approximately twenty seconds on the Cardano blockchain).
+
+By parameterising transaction validity by the blockchain information
+generating function $\sigma$, we obtain an equivalence class of
+transactions that are indistinguishable by validator scripts with
+access to the result of $\sigma$.
+
+\subsection{Examples}
+\blue{It would be good to have one or more examples here, especially one
+  demonstrating what the separation of redeemer and datascript does.}
+
+\newpage
+\section{XUTXO-2: multicurrency support}
+\label{sec:xutxo-2}
+We now extend the XUTXO model to allow transactions to deal with
+multiple currencies.  We assume that the underlying blockchain has a
+unique \textit{native currency} which is the fundamental currency used
+for paying fees, for example.  The native currency of the Cardano
+blockchain is Ada.
+
+In addition to the native currency, the XUTXO-2 model allows an
+unlimited number of other \textit{custom currencies}, possibly
+user-defined.  Each custom currency has a unique identifier; the
+native currency is handled separately and has no identifier.  To
+support custom currencies we introduce three new types: see
+Figure~\ref{fig:more-basic-types}.
```

Comment:
Maybe we should mention the native currency here, but not write it down in the spec - it's only needed for the fee, we don't have a cost model for XUTXO yet, and by ignoring the native currency in the definitions we could delete the `forge`, `fee` and `value` fields

Reply:
Yes, I wasn't sure about that.  I separated the custom currencies out from the native currency because it seemed that they really are different:  you have fees for the native currency, but not for the custom ones, and in Cardanomy feeling is that Ada really is the basic currency that underlies everything else.  I think I'll leave it the way it is for now and see what Manuel says.